intel_extension_for_transformers/llm/runtime/graph/core/ne_layers.c

```diff
@@ -2778,7 +2778,7 @@ struct ne_tensor* ne_soft_max_inplace(struct ne_context* ctx, struct ne_tensor*
 // ne_rope
 
 struct ne_tensor* ne_rope_impl(struct ne_context* ctx, struct ne_tensor* a, int n_past, int n_dims, int mode,
-                               bool inplace) {
+                               int n_ctx, bool inplace) {
```

Comment:
I don't know if it is a good idea to add an extra input parameter `n_ctx` here. BTW, `n_ctx` can be get by `ctx->model.hparams.n_ctx`

Reply:
The ggml rope has two speical branch for gptjneox and chatglm1.
https://github.com/ggerganov/ggml/blob/master/src/ggml.c#L12781

We use the ggml rope kernel. maybe we should keep the same as ggml. don't know the reason why delete the glm branch before.

Reply:
The rope requires the size of prompts but not the fixed model.haprams.n_ctx.